src/System.Private.ServiceModel/tests/Scenarios/Binding/Http/BasicHttpBindingTests.cs

```diff
@@ -4,17 +4,20 @@
 using System;
 using System.ServiceModel;
 using System.ServiceModel.Channels;
+using System.ServiceModel.Tests.Common;
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
 using TestTypes;
 using Xunit;
 
-public static class Binding_Http_BasicHttpBindingTests
+public class Binding_Http_BasicHttpBindingTests : OuterLoopTests
```

Comment:
I think I like your idea of having a class called `Endpoints`  and it could abstract abstract the parsing and handling of return Json etc and the callsite only gets back a valid URL.  

```csharp
 [OuterLoop]
public static void DefaultSettings_Echo_RoundTrips_String()
{
   string variationDetails = "Client:: NetHttpBinding/DefaultValues\nServer:: NetHttpBinding/DefaultValues";

    ChannelFactory<IWcfService> factory = 
                          new ChannelFactory<IWcfService>(binding, new 
                                 EndpointAddress(Endpoints.HttpBaseAddress_NetHttp));
}
```
But constructing the URL using `"/Basic"` seems odd since the server returns a part of the URL and the client builds the rest. I would have expected the server to return an array of endpoints and you pick out one based on a name. This would avoid issues of hitting a wrong service a non-existent endpoint for the service. Find could be an extension to search the returned array type

```csharp
public static class Endpoints {

#region wcfservice endpoints 
    // Return the endpoint with matching names.
    // Find might throw an Invalid Endpoint name 
    public static EndpointAddress HttpBaseAddress_Basic{
          return GetEndpoint("wcfservice", "basic");
    }

private static Url GetEndpoint(string servicename, string endpointname)
{
       var endpoints = BridgeClient.GetOrCreateService(servicename);
       var endpoint = endpoints.FirstOrDefault(e => String.Compare(e.name,endpointname,true) == 0);
       if(endpoint != null){
           return new Url(endpoint.ListenUri);
       }

      throw new Exception("Endpoint not initialized.");
}
#endregion 
}
```